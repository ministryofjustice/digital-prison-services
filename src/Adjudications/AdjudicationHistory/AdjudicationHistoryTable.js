import React from 'react'
import PropTypes from 'prop-types'
import Table from '@govuk-react/table'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import pathToRegexp from 'path-to-regexp'
import routePaths from '../../routePaths'
import { Flag } from '../../flags'

import { LargeScreenOnly, SmallScreenOnly } from '../../Components/sizing'

import PreviousNextNavigation from '../../PreviousNextNavigation'
import ResultsFilter from '../../Components/ResultsFilter'

const adjudicationPath = pathToRegexp.compile(routePaths.adjudication)

const AdjudicationLink = ({ offenderNo, adjudicationNo }) => (
  <Flag
    name={['adjudicationDetailsLinkEnabled']}
    render={() => (
      <Link to={adjudicationPath({ offenderNo, adjudicationNo })} className="link">
        {adjudicationNo}
      </Link>
    )}
    fallbackRender={() => <>{adjudicationNo}</>}
  />
)

AdjudicationLink.propTypes = {
  offenderNo: PropTypes.string.isRequired,
  adjudicationNo: PropTypes.number.isRequired,
}

export const AdjudicationHistoryTable = ({
  offenderNo,
  results,
  totalRecords,
  pageSize,
  pageNumber,
  changePerPage,
  changePage,
}) => (
  <>
    <ResultsFilter perPage={pageSize} pageNumber={pageNumber} totalResults={totalRecords} noBorder>
      <ResultsFilter.PerPageDropdown handleChange={changePerPage} totalResults={totalRecords} perPage={pageSize} />
    </ResultsFilter>

    <LargeScreenOnly>
      <Table
        className="results"
        head={
          <Table.Row>
            <Table.CellHeader setWidth="10%">Adjudication number</Table.CellHeader>
            <Table.CellHeader setWidth="15%">Report date and time</Table.CellHeader>
            <Table.CellHeader setWidth="20%">Establishment</Table.CellHeader>
            <Table.CellHeader setWidth="45%">Offence description</Table.CellHeader>
            <Table.CellHeader setWidth="10%">Finding</Table.CellHeader>
          </Table.Row>
        }
      >
        {results.map(result => (
          <Table.Row key={result.adjudicationNumber}>
            <Table.Cell>
              <AdjudicationLink offenderNo={offenderNo} adjudicationNo={result.adjudicationNumber} />
            </Table.Cell>
            <Table.Cell>{`${result.reportDate} ${result.reportTime}`}</Table.Cell>
            <Table.Cell>{result.establishment}</Table.Cell>
            <Table.Cell>{result.offenceDescription}</Table.Cell>
            <Table.Cell>{result.findingDescription}</Table.Cell>
          </Table.Row>
        ))}
      </Table>
    </LargeScreenOnly>
    <SmallScreenOnly>
      <Table
        className="results"
        head={
          <Table.Row>
            <Table.CellHeader>Adjudication Number</Table.CellHeader>
            <Table.CellHeader>Report date</Table.CellHeader>
            <Table.CellHeader>Finding</Table.CellHeader>
          </Table.Row>
        }
      >
        {results.map(result => (
          <Table.Row key={result.adjudicationNumber}>
            <Table.Cell>
              <AdjudicationLink offenderNo={offenderNo} adjudicationNo={result.adjudicationNumber} />
            </Table.Cell>
            <Table.Cell>{result.reportDate}</Table.Cell>
            <Table.Cell>{result.findingDescription}</Table.Cell>
          </Table.Row>
        ))}
      </Table>
    </SmallScreenOnly>

    <div>
      <PreviousNextNavigation
        pagination={{ perPage: pageSize, pageNumber }}
        totalRecords={totalRecords}
        pageAction={changePage}
      />
    </div>
  </>
)

AdjudicationHistoryTable.propTypes = {
  totalRecords: PropTypes.number.isRequired,
  pageSize: PropTypes.number.isRequired,
  pageNumber: PropTypes.number.isRequired,
  offenderNo: PropTypes.string.isRequired,
  results: PropTypes.arrayOf(
    PropTypes.shape({
      adjudicationNumber: PropTypes.number.isRequired,
      reportDate: PropTypes.string.isRequired,
      reportTime: PropTypes.string.isRequired,
      establishment: PropTypes.string.isRequired,
      offenceDescription: PropTypes.string.isRequired,
      findingDescription: PropTypes.string,
    })
  ).isRequired,
  changePerPage: PropTypes.func.isRequired,
  changePage: PropTypes.func.isRequired,
}

const mapStateToProps = state => ({
  results: state.adjudicationHistory.results,
  totalRecords: state.adjudicationHistory.totalRecords,
  pageSize: state.adjudicationHistory.pageSize,
  pageNumber: state.adjudicationHistory.pageNumber,
})

export default connect(mapStateToProps)(AdjudicationHistoryTable)
