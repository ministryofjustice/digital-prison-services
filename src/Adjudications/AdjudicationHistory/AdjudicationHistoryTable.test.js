import React from 'react'
import testRenderer from 'react-test-renderer'

import AdjudicationHistoryTable from './AdjudicationHistoryTable'

const initialState = {
  adjudicationHistory: {
    pageNumber: 0,
    pageSize: 20,
    totalRecords: 100,
    results: [
      {
        adjudicationNumber: 1111111,
        reportDate: '12/03/2018',
        reportTime: '12:20',
        establishment: 'Moorland',
        offenceDescription: 'Arson',
        findingDescription: 'Guilty',
      },
      {
        adjudicationNumber: 2222222,
        reportDate: '17/02/2017',
        reportTime: '10:20',
        establishment: 'Leeds',
        offenceDescription: 'Assault',
        findingDescription: ' Not Guilty',
      },
      {
        adjudicationNumber: 3333333,
        reportDate: '01/03/2017',
        reportTime: '04:20',
        establishment: 'Moorland',
        offenceDescription: 'Theft',
        findingDescription: '--',
      },
    ],
  },
}

describe('Adjudication History Table', () => {
  const store = {}
  const history = {}
  beforeEach(() => {
    store.getState = jest.fn()
    store.subscribe = jest.fn()
    store.dispatch = jest.fn()
    history.push = jest.fn()
    history.replace = jest.fn()
    store.getState.mockReturnValue(initialState)
  })

  it('should render correctly', () => {
    const wrapper = testRenderer
      .create(<AdjudicationHistoryTable store={store} changePerPage={jest.fn()} changePage={jest.fn()} />)
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
