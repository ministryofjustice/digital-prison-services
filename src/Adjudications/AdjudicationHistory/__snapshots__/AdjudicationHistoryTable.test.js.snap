// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adjudication History Table should render correctly 1`] = `
Array [
  <div
    className="sc-jzJRlG doLUQE"
  >
    <div
      className="sc-fjdhpX gPQCpk"
    >
      1
       - 
      20
       of
       
      100
       results
    </div>
    <div
      className="sc-kpOJdX gBCKOX"
    >
      <label
        htmlFor="perPage"
      >
        <span
          className="sc-cSHVUG lowffx"
        >
          Results per page
        </span>
      </label>
      <select
        className="sc-kGXeez eEExxo"
        id="perPage"
        onChange={[Function]}
        value={20}
      >
        <option
          value="10"
        >
          10
        </option>
        <option
          value="20"
        >
          20
        </option>
        <option
          value="50"
        >
          50
        </option>
        <option
          value={100}
        >
          All
        </option>
      </select>
    </div>
  </div>,
  <div
    className="sc-gzVnrw gIvEvv"
  >
    <table
      className="sc-EHOje ecdksn results"
    >
      <thead
        className="sc-bdVaJa bDWFJH"
      >
        <tr
          className="sc-ifAKCX cFlEyZ"
        >
          <th
            className="sc-bxivhb fWvmaq"
          >
            Adjudication number
          </th>
          <th
            className="sc-bxivhb hgtSCp"
          >
            Report date and time
          </th>
          <th
            className="sc-bxivhb dnVQUz"
          >
            Establishment
          </th>
          <th
            className="sc-bxivhb blvPUL"
          >
            Offence description
          </th>
          <th
            className="sc-bxivhb fWvmaq"
          >
            Finding
          </th>
        </tr>
      </thead>
      <tbody
        className="sc-bwzfXH uTxCW"
      >
        <tr
          className="sc-ifAKCX cFlEyZ"
        >
          <td
            className="sc-bxivhb ckLzJk"
          >
            <a
              className="link"
              href="/offenders/AAA123/adjudications/1111111"
              onClick={[Function]}
            >
              1111111
            </a>
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            12/03/2018 12:20
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            Moorland
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            Arson
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            Guilty
          </td>
        </tr>
        <tr
          className="sc-ifAKCX cFlEyZ"
        >
          <td
            className="sc-bxivhb ckLzJk"
          >
            <a
              className="link"
              href="/offenders/AAA123/adjudications/2222222"
              onClick={[Function]}
            >
              2222222
            </a>
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            17/02/2017 10:20
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            Leeds
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            Assault
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
             Not Guilty
          </td>
        </tr>
        <tr
          className="sc-ifAKCX cFlEyZ"
        >
          <td
            className="sc-bxivhb ckLzJk"
          >
            <a
              className="link"
              href="/offenders/AAA123/adjudications/3333333"
              onClick={[Function]}
            >
              3333333
            </a>
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            01/03/2017 04:20
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            Moorland
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            Theft
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            --
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="sc-bZQynM iVQEAa"
  >
    <table
      className="sc-EHOje ecdksn results"
    >
      <thead
        className="sc-bdVaJa bDWFJH"
      >
        <tr
          className="sc-ifAKCX cFlEyZ"
        >
          <th
            className="sc-bxivhb eWDuGm"
          >
            Adjudication Number
          </th>
          <th
            className="sc-bxivhb eWDuGm"
          >
            Report date
          </th>
          <th
            className="sc-bxivhb eWDuGm"
          >
            Finding
          </th>
        </tr>
      </thead>
      <tbody
        className="sc-bwzfXH uTxCW"
      >
        <tr
          className="sc-ifAKCX cFlEyZ"
        >
          <td
            className="sc-bxivhb ckLzJk"
          >
            <a
              className="link"
              href="/offenders/AAA123/adjudications/1111111"
              onClick={[Function]}
            >
              1111111
            </a>
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            12/03/2018
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            Guilty
          </td>
        </tr>
        <tr
          className="sc-ifAKCX cFlEyZ"
        >
          <td
            className="sc-bxivhb ckLzJk"
          >
            <a
              className="link"
              href="/offenders/AAA123/adjudications/2222222"
              onClick={[Function]}
            >
              2222222
            </a>
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            17/02/2017
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
             Not Guilty
          </td>
        </tr>
        <tr
          className="sc-ifAKCX cFlEyZ"
        >
          <td
            className="sc-bxivhb ckLzJk"
          >
            <a
              className="link"
              href="/offenders/AAA123/adjudications/3333333"
              onClick={[Function]}
            >
              3333333
            </a>
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            01/03/2017
          </td>
          <td
            className="sc-bxivhb ckLzJk"
          >
            --
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div>
    <nav
      className="sc-htoDjs kGbKV"
    >
      <ul
        className="sc-dnqmqq gfICuL"
      >
        <li
          className="sc-iwsKbI dBJiGn"
        />
        <li
          className="sc-gqjmRU fcaAIL"
          id="next-page"
          onClick={[Function]}
          role="button"
        >
          <div
            className="sc-gZMcBi ckrsaC"
          >
            Next
            <span
              className="sc-VigVT fTVswn"
            >
              page
            </span>
            <span
              className="sc-jTzLTM gFPDJc"
            >
              2
               of 
              5
            </span>
          </div>
        </li>
      </ul>
    </nav>
  </div>,
]
`;
